### PR TITLE
fix for bugs #135 and #136

### PR DESCRIPTION
**Pull request by singularis-mzf**
This is my fix for the bugs. I tested that the bugs were fixed by this, but I didn't run designed tests for the mod.

### CODE REVIEW

File: ui/compose.lua [hunk 1]
```diff
@@ -40,10 +38,7 @@ minetest.register_on_player_receive_fields(function(player, formname, fields)
 
 	local name = player:get_player_name()
     if fields.send then
-        local id = mail.new_uuid()
-        if mail.selected_idxs.message[name] then
-            id = mail.selected_idxs.message[name]
-        end
+        local id = mail.selected_idxs.message[name] or mail.new_uuid()
```
Comment (S-S-X) [inline ui/compose.lua:41]:
```suggestion
	local id = mail.selected_idxs.message[name] or mail.new_uuid()
```
Yeah, I know it is all mixed spaces and tabs but hoping to slowly move towards all tabs indentation.

**Review by S-S-X — APPROVED** [review]
I do not really know code base well enough to be sure but still I believe for mail.show_compose this is correct behavior: generate new id for selection if no id available yet.

This is for new mail right? And existing id is for drafts?

**Review by Athozus — APPROVED** [review]
LGTM. Didn't tested but 99% seems ok according to my knowledge of the process of this file.

**Comment by Athozus** [comment]
> I do not really know code base well enough to be sure but still I believe for mail.show_compose this is correct behavior: generate new id for selection if no id available yet.
> 
> This is for new mail right? And existing id is for drafts?

Yes, the thing is to distinguish previously blank messages and draft messages being sent.

**Pull request merged**